test/jdk/sun/net/ext/ExtendedSocketOptionsTest.java

```diff
@@ -0,0 +1,94 @@
+/*
+ * Copyright (c) 2021, Oracle and/or its affiliates. All rights reserved.
+ * DO NOT ALTER OR REMOVE COPYRIGHT NOTICES OR THIS FILE HEADER.
+ *
+ * This code is free software; you can redistribute it and/or modify it
+ * under the terms of the GNU General Public License version 2 only, as
+ * published by the Free Software Foundation.  Oracle designates this
+ * particular file as subject to the "Classpath" exception as provided
+ * by Oracle in the LICENSE file that accompanied this code.
+ *
+ * This code is distributed in the hope that it will be useful, but WITHOUT
+ * ANY WARRANTY; without even the implied warranty of MERCHANTABILITY or
+ * FITNESS FOR A PARTICULAR PURPOSE.  See the GNU General Public License
+ * version 2 for more details (a copy is included in the LICENSE file that
+ * accompanied this code).
+ *
+ * You should have received a copy of the GNU General Public License version
+ * 2 along with this work; if not, write to the Free Software Foundation,
+ * Inc., 51 Franklin St, Fifth Floor, Boston, MA 02110-1301 USA.
+ *
+ * Please contact Oracle, 500 Oracle Parkway, Redwood Shores, CA 94065 USA
+ * or visit www.oracle.com if you need additional information or have any
+ * questions.
+ */
+
+import org.testng.Assert;
+import org.testng.annotations.Test;
+
+import java.util.concurrent.Callable;
+import java.util.concurrent.ExecutorService;
+import java.util.concurrent.Executors;
+import java.util.concurrent.Future;
+
+/**
+ * @test
+ * @bug 8260366
+ * @summary Verify that concurrent classloading of sun.net.ext.ExtendedSocketOptions and
+ * jdk.net.ExtendedSocketOptions doesn't lead to a deadlock
+ * @run testng/othervm --add-exports=java.base/sun.net.ext=ALL-UNNAMED ExtendedSocketOptionsTest
```

Comment:
The --add-exports should not be needed - have you tried simply adding:

```
@modules java.base/sun.net.ext:+open
                  jdk.net
```

before `@run`?  This also has the additional benefit to declare which modules are required to run the test.

Reply:
Since the test should be fast, I'd also advise to stick several identical @run lines (maybe e.g. 5 of them) to increase the probability of the deadlock to happen...

Reply:
> Since the test should be fast, I'd also advise to stick several identical @run lines (maybe e.g. 5 of them) to increase the probability of the deadlock to happen...

Done. The PR has been updated to run this test multiple times now.

Reply:
> The --add-exports should not be needed - have you tried simply adding:
> 
> ```
> @modules java.base/sun.net.ext:+open
>                   jdk.net
> ```
> 
> before `@run`? This also has the additional benefit to declare which modules are required to run the test.

I had forgotten about the `@modules` directive. I have now updated the PR use that instead. The only minor difference between what you suggested and my updated PR is that I decided to use `:open` instead of `:+open` for the `sun.net.ext` package, since I don't use the types in that package at compile time, in that test.